File: DSA/RATNA_KHANNA_590013141/DAY_50_QUESTION_3.c
#include <stdio.h>

int main() {
    int n, target;
     printf("enter number of elements\n");
    scanf("%d", &n);
    int arr[n];
    printf("enter elements\n");
    for(int i = 0; i < n; i++)
        scanf("%d", &arr[i]);
    printf("enter target\n");
    scanf("%d", &target);

    int start = 0, sum = 0, found = 0;
    for(int end = 0; end < n; end++) {
        sum += arr[end];
        while(sum > target && start < end)
            sum -= arr[start++];

        if(sum == target) {
            printf("Subarray found from index %d to %d", start, end);
            found = 1;
            break;
        }
    }

    if(!found)
        printf("No subarray found");
    return 0;
}

File: DSA/AnshulJain_590014934/Day_50_Question_3.py
arr = list(map(int, input("Enter the array:-").split()))
target = int(input("Enter the target sum: "))
start = 0
current_sum = 0
found = False
for end in range(len(arr)):
    current_sum += arr[end]
    while current_sum > target and start < end:
        current_sum -= arr[start]
        start += 1
    if current_sum == target:
        print(f"Subarray found from index {start} to {end}")
        found = True
        break
if not found:
    print("No subarray found")
#for example:
# Input: Enter the array:- 1 2 3 4 5
#        Enter the target sum: 9
# Output: Subarray found from index 0 to 3  
